components/eamxx/cime_config/eamxx_buildnml.py

```diff
@@ -238,15 +246,16 @@ def evaluate_selectors(element, case, ez_selectors):
     child_values = {} # elem_name -> evaluated XML element
     children_to_remove = []
     for child in element:
-        child_name = child.tag
-        child_val = child.text
-
         # Note: in our system, an XML element is either a "node" (has children)
         # or a "leaf" (has a value).
         has_children = len(child) > 0
         if has_children:
             evaluate_selectors(child, case, ez_selectors)
         else:
+            child_name = child.tag
+            child.text = None if child.text is None else child.text.strip(' \n')
```

Comment:
Text should never be `None` now, right?

Reply:
If there is a parameter like `<my_param/>` (open and closed in one set of brackets), then the text will indeed be set to None by the XML parser.

Reply:
@bartgol 

That's is unfortunate.

I thought `<x/>` was just semantic sugar for `<x></x>`. Are you saying the former gives text=None and the latter gives text=""?

Reply:
I was about to say "yes", then I checked with this mini script
```python
import xml.etree.ElementTree as ET


s = """ 
<tree>
    <a/>
    <b></b>
</tree>
"""
xml = ET.fromstring(s)

for a in xml:
    print(f"{a.tag} = {a.text}")
```
which printed
```bash
a = None
b = None
```
so now I'm not sure anymore...